### PR TITLE
Allow stopping a single process while running

### PR DESCRIPTION
**Pull request by s7932**
Pull Request Description:
This PR adds a new feature to the application by implementing a button that allows users to stop a single process during runtime using a specific technique. Additionally, it includes handling for the scenario where official processes cannot be terminated.

Changes Made:
Added a new button to the UI for stopping a single process during runtime.
Implemented the functionality to stop a process using the specified technique.
Added logic to prevent the termination of official processes.

Checklist:
 The code follows the project's coding style guidelines.
 Unit tests have been added to cover the new functionality.
 Documentation has been updated to reflect the changes.
 The feature has been tested thoroughly.

### CODE REVIEW

File: gui/CMakeLists.txt [hunk 1]
```diff
@@ -54,6 +54,54 @@ target_link_libraries(DraggableSquares PRIVATE ${BSON_LIBRARIES} Qt5::Core Qt5::
...
+#     src/main_window.cpp
+#     src/simulation_data_manager.cpp
+# )
+
+# # Include BSON and project directories
+# target_include_directories(DraggableSquares PRIVATE ${BSON_INCLUDE_DIRS} include)
+
+# # Link the BSON library and Qt5 components
+# target_link_libraries(DraggableSquares PRIVATE ${BSON_LIBRARIES} Qt5::Core Qt5::Gui Qt5::Widgets)
+
```
Comment (ChedvaErblich) [inline gui/CMakeLists.txt]: Why did you comment out this code?

File: gui/include/frames.h [hunk 2]
```diff
@@ -22,7 +22,7 @@ class Frames : public QWidget {
 
     // Ctor
     Frames(LogHandler &logHandler, QWidget *parent = nullptr);
-
+    ~Frames() {};
```
Comment (ChedvaErblich) [inline gui/include/frames.h]: If you use an empty d-tor, consider changing ~Manager(){} to ~Manager() = default;. This uses the default destructor generated by the compiler, which is generally sufficient if there is no custom cleanup logic

File: gui/src/draggable_square.cpp [hunk 3]
```diff
@@ -30,17 +30,20 @@ void DraggableSquare::setSquareColor(const QString &color)
 {
     setStyleSheet(color);
 }
-//constructor
+// Update the constructor definition to match the declaration
```
Comment (ChedvaErblich) [inline gui/src/draggable_square.cpp]: Is it left out?

File: gui/src/process_dialog.cpp [hunk 4]
```diff
@@ -58,6 +58,7 @@ ProcessDialog::ProcessDialog(QWidget *parent) : QDialog(parent)
 
     setLayout(layout);
 }
+ProcessDialog::~ProcessDialog() {}
```
Comment (ChedvaErblich) [inline gui/src/process_dialog.cpp]: Same comment as above,
regarding the D-TOR...

File: gui/src/simulation_data_manager.cpp [hunk 5]
```diff
@@ -13,6 +13,7 @@
 SimulationDataManager::SimulationDataManager(QWidget *parent) : QWidget(parent)
 {
 }
+SimulationDataManager::~SimulationDataManager() {}
```
Comment (ChedvaErblich) [inline gui/src/simulation_data_manager.cpp]: Same comment as above,
regarding the D-TOR...

**Comment by ChedvaErblich** [comment]
I understand the "nimas" 😉
But please update the commit message, or if there's no reason to split it into two commits, just squash them together.

Also, it would be great to have a way to see when a process has stopped.
We can disable the stop button if the process is already stopped. I mean, the button should remain visible but be unclickable.

File: gui/CMakeLists.txt (hunk 1 above)
Comment (s7932) [reply gui/CMakeLists.txt]: Indeed, these lines should not appear in the code and were removed with the commit that was raised by mistake.

File: gui/include/frames.h (hunk 2 above)
Comment (s7932) [reply gui/include/frames.h]: Indeed, these lines should not appear in the code and were removed with the commit that was raised by mistake.

File: gui/src/draggable_square.cpp (hunk 3 above)
Comment (s7932) [reply gui/src/draggable_square.cpp]: fixed :)

File: gui/src/process_dialog.cpp (hunk 4 above)
Comment (s7932) [reply gui/src/process_dialog.cpp]: Indeed, these lines should not appear in the code and were removed with the commit that was raised by mistake.

File: gui/src/simulation_data_manager.cpp (hunk 5 above)
Comment (s7932) [reply gui/src/simulation_data_manager.cpp]: Indeed, these lines should not appear in the code and were removed with the commit that was raised by mistake.

File: gui/src/main_window.cpp [hunk 6]
```diff
@@ -693,4 +747,26 @@ void MainWindow::createProcessConfigFile(int id, const QString &processPath)
...
+
+void MainWindow::processFinished(int exitCode, QProcess::ExitStatus exitStatus)
+{
+    QProcess *finishedProcess = qobject_cast<QProcess *>(sender());
+    if (finishedProcess) {
+        // חפשי את ה-ID של התהליך שסיים
+        for (const QPair<QProcess*, int>& pair : runningProcesses) {
+            if (pair.first == finishedProcess) {
+                int finishedProcessId = pair.second;
+                // מצאי את ה-DraggableSquare המתאים
```
Comment (ChedvaErblich) [inline gui/src/main_window.cpp]: Please write comments in English :)

Comment (s7932) [reply gui/src/main_window.cpp]: fixed!!

**Pull request merged**